server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedDecoder.java

```diff
@@ -106,13 +109,17 @@ public MemcachedDecoder(AdvancedCache<byte[], byte[]> memcachedCache, ScheduledE
       this.ignoreCache = ignoreCache;
       isStatsEnabled = cache.getCacheConfiguration().statistics().enabled();
       this.timeService = memcachedCache.getComponentRegistry().getTimeService();
+      BlockingManager blockingManager = SecurityActions.getGlobalComponentRegistry(memcachedCache.getCacheManager())
+            .getComponent(BlockingManager.class);
+      responseExecutor = new LimitedExecutor("memcached-sender", blockingManager.asExecutor("memcached-sender"), 1);
    }
 
    private final AdvancedCache<byte[], byte[]> cache;
    private final ScheduledExecutorService scheduler;
    protected final NettyTransport transport;
    protected final Predicate<? super String> ignoreCache;
    private final TimeService timeService;
+   protected final Executor responseExecutor;
```

Comment:
The `lastResponse` variable below doesn't need volatile now either as it is only reference from the event loop.